src/mpi/coll/src/coll_impl.c

```diff
@@ -18,22 +18,23 @@
 cvars:
     - name        : MPIR_CVAR_DEVICE_COLLECTIVES
       category    : COLLECTIVE
-      type        : boolean
-      default     : true
-      class       : device
+      type        : enum
+      default     : percoll
+      class       : none
       verbosity   : MPI_T_VERBOSITY_USER_BASIC
       scope       : MPI_T_SCOPE_ALL_EQ
-      description : >-
-        If set to true, MPI collectives will allow the device to override
-        the MPIR-level collective algorithms. The device still has the
-        option to call the MPIR-level algorithms manually.  If set to false,
-        the device-level collective function will not be called.
+      description : |-
+        Variable to select whether the device can override the
+        collective algorithm.
+        all     - Always prefer the device collectives
+        none    - Never pick the device collectives
+        percoll - Use the per-collective CVARs to decide
 
     - name        : MPIR_CVAR_COLLECTIVE_FALLBACK
       category    : COLLECTIVE
       type        : int
       default     : 0
-      class       : device
+      class       : none
       verbosity   : MPI_T_VERBOSITY_USER_BASIC
       scope       : MPI_T_SCOPE_ALL_EQ
       description : >-
```

Comment:
Why don't we turn this one into a enum -- e.g. error, print, fallback?

Reply:
Good idea.  Fixed.